lib/components/molecules/Search/Search.js

```diff
@@ -0,0 +1,111 @@
+// @flow
+/**
+ *
+ * Search
+ *
+ */
+import React, { useState, Node, useCallback } from 'react';
+import styled from 'styled-components';
+import classnames from 'classnames';
+
+import { useSpeechRecognition } from '../../../core/hooks';
+import Input from '../../atoms/Input';
+import Button from '../../atoms/Button';
+import SearchList from './SearchList/SearchList';
+import styles from './Search.style';
+import type { Props } from './types';
+
+const Search = ({ className, inputProps, dataLists, renderListItem }: Props): Node => {
+  const [inputValue, setInputValue] = useState(inputProps.value || '');
+  const [searchLists, setSearchList] = useState([]);
+
+  const onSearchClick = useCallback(
+    searchText => {
+      const searchValue = typeof searchText === 'string' ? searchText : inputValue;
+
+      if (searchValue) {
+        const lists = dataLists.filter(({ name }) => {
+          return name.toLowerCase().indexOf(searchValue.toLowerCase()) !== -1;
+        });
+        setSearchList(lists);
+      }
+      return false;
+    },
+    [inputValue, dataLists, setSearchList]
+  );
+
+  const onResult = result => {
```

Comment:
Can we make onResult more explicit? its ambiguous what it does 

onSpeachRecognitionResults makes it more clear